### DISCUSSION TITLE
Read-only volumes

### DISCUSSION THREAD

**Discussion by brandonkelly** [craftcms/cms#8464] [Ideas]
Similar to the “Assets in this volume have public URLs” volume setting, volumes should have a “Read-only” setting, which tells Craft not to allow new files to be uploaded/moved to the volume.

**Comment by johndwells** [comment]
This would be particularly helpful when it comes to remote volumes + multiple environments (production, staging, development, etc).

**Comment by sa3dany** [comment]
A workaround is to set the [`maxUploadFileSize`](https://docs.craftcms.com/v3/config/config-settings.html#maxuploadfilesize) config variable to `1` (0 is unlimited) in environments where you want to prevent uploads. This only works if you want to disable uploads across all volumes though.
